### PR TITLE
remove bbc-a11y iframe exceptions

### PR DESCRIPTION
**Pull request by jroebu14**
Resolves #NUMBER

**Overall change:**
Remove bbc-a11y iframe exceptions as main headings in iframes is not a requirement anymore. This was fixed in bbc-a11y@2.4.2

**Code changes:**

- Upgrade to bbc-a11y@2.4.2
- Remove iframe exceptions from bbc-a11y.js
---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by karinathomasbbc — APPROVED** [review]

File: a11y.js [hunk 1]
```diff
@@ -27,11 +27,9 @@ const pageTypes = {
   mostReadPage: ["//div[@id='root']/header/nav/div/div[1]/div/ul"],
   storyPage: [
     "//div[@id='root']/header/nav/div/div[1]/div/ul",
-    '/iframe', // known issue above with iframes should be revisited once https://github.com/bbc/bbc-a11y/issues/298 is resolved
     "//div[@id='root']/div/div[1]/main/div[37]/div/div/div", // issue with IDT2 includes
   ],
   onDemandRadio: [
-    '/iframe', // Can be removed once https://github.com/bbc/bbc-a11y/issues/298 is resolved
     "//div[@id='root']/main/div/div/div/iframe",
```
Comment (greenc05) [inline a11y.js]: And this?

Comment (greenc05) [reply a11y.js]: And line 23 for live radio?

Comment (jroebu14) [reply a11y.js]: Sorted, thanks

**Review by greenc05 — APPROVED** [review]
Looks good assuming you've ran bbc-a11y on the relevant pages and nothing new is failing....

File: a11y.js [hunk 2]
```diff
@@ -9,30 +9,51 @@ const isSmoke = true;
...
+  // * Editorial: Indicating language: Html must have lang attribute
+  'html tag has no lang attribute: /html',
+
+  // * Forms: Labelling form controls: Fields must have labels or titles
+  "Button has no text: //button[@id='p_audioui_playpause']",
+  "Button has no text: //button[@id='p_audioui_previousButton']",
+  "Button has no text: //button[@id='p_audioui_nextButton']",
+  "Button has no text: //div[@id='p_audioui_container']/div[2]/button",
+  "Button has no text: //button[@id='p_audioui_playbackSettingsButton']",
+  "Button has no text: //div[@id='p_audioui_container']/div[4]/button",
```
Comment (jroebu14) [inline a11y.js:39]: @greenc05 all of these are failing when I removed the full iframe error suppression. Do you think we should have the error suppressors more specific so we can catch stuff we can fix like the html language one and potentially the button ones? We could also see if moving to Toucan means we can remove some of these error suppressions.

Comment (greenc05) [reply a11y.js:39]: I'm not sure, would that make it more fragile if SMP update their code?

Comment (greenc05) [reply a11y.js:39]: Maybe we should just put them back and wait for Toucan?

**Review by pharingee — APPROVED** [review]

File: a11y.js (hunk 2 above)
Comment (jroebu14) [reply a11y.js:39]: If they change their code and cause more issues then yeh this would fail

Comment (greenc05) [reply a11y.js:39]: Cool, lets revert for pages where we got these errors then, can re-look when we get Toucan on pages with SMP

**Pull request merged**